dbt-snowflake/src/dbt/adapters/snowflake/impl.py

```diff
@@ -273,37 +278,69 @@ def get_catalog_for_single_relation(
             stats=stats_dict,
         )
 
+    def enable_iceberg_materializations_to_reconstruct_is_iceberg(self, schema):
+        """
+        enable_iceberg_materializations == True:
+            bundle is out of date and lacks `is_iceberg` on SHOW OBJECTS.
+            Advisory: this adds a latency penalty for the snowflake users when creating
+            tables. This only affects, however, those that refuse to upgrade to new
+            bundle by 4/28, which should be very few since the new bundle is required.
+        enable_iceberg_materializations == False:
+            `is_iceberg` now comes for free in warehouse
+
+        Will only ever execute flag change once. I tried to use lru_cache but couldn't because
+        self is not hashable.
+        """
+
+        func = type(self).enable_iceberg_materializations_to_reconstruct_is_iceberg
+
+        if getattr(func, "_has_run", False):
+            return
+
+        try:
+            _, objs = self.execute(f"show objects in {schema} limit 1", fetch=True)
+        except DbtDatabaseError as exc:
+            if "002043 (02000)" in str(exc):
+                return  # a la 'rollback transaction'
+
+        # Abuse of the behavior flag interface to overwrite it / change its semantics
+        # to a flag like: is there is_iceberg or must we reconstruct it?
+        # Don't do this. My hand was forced because of need of a schema.
```

Comment:
This is solution no. 4. I did not reach for this at first and actively tried to avoid it. The placement of this function makes it quite hard to efficiently determine behavior dependent on metadata schema during the runtime